### PR TITLE
warning messages fixed

### PR DESCRIPTION
**Pull request by pvmm**

### CODE REVIEW

File: src/MSXMotherBoard.cc [hunk 1]
```diff
@@ -705,8 +705,8 @@ MSXMapperIO& MSXMotherBoard::createMapperIO()
 
 		MSXCPUInterface& cpuInterface = getCPUInterface();
 		for (auto port : {0xfc, 0xfd, 0xfe, 0xff}) {
```
Comment (pvmm) [inline src/MSXMotherBoard.cc:707]: Even though values fit in a byte, compiler still complains about lost precision, so I used the narrow_cast.
```
src/MSXMotherBoard.cc:707:18: warning: implicit conversion loses integer precision: 'const int' to 'byte' (aka 'unsigned char') [-Wimplicit-int-conversion]
  707 |                 for (byte port : {0xfc, 0xfd, 0xfe, 0xff}) {
      |                           ~~~~ ^
```

File: src/MSXMotherBoard.cc [hunk 2]
```diff
@@ -721,8 +721,8 @@ void MSXMotherBoard::destroyMapperIO()
 	if (mapperIOCounter == 0) {
 		MSXCPUInterface& cpuInterface = getCPUInterface();
 		for (auto port : {0xfc, 0xfd, 0xfe, 0xff}) {
```
Comment (pvmm) [inline src/MSXMotherBoard.cc:723]: Even though values fit in a byte, compiler still complains about lost precision, so I used the narrow_cast.
```
src/MSXMotherBoard.cc:707:18: warning: implicit conversion loses integer precision: 'const int' to 'byte' (aka 'unsigned char') [-Wimplicit-int-conversion]
  707 |                 for (byte port : {0xfc, 0xfd, 0xfe, 0xff}) {
      |                           ~~~~ ^
```

**Comment by m9710797** [comment]
Hi. Thanks!
I agree with your change. I'll merge this PR later when I'm at home.

But I'm curious: which compiler (gcc, clang, ...) and which version are you using? And which build flavor (debug, devel, opt, ...)?
Because I did not see these warnings.

**Comment by pvmm** [comment]
> Hi. Thanks! I agree with your change. I'll merge this PR later when I'm at home.
> 
> But I'm curious: which compiler (gcc, clang, ...) and which version are you using? And which build flavor (debug, devel, opt, ...)? Because I did not see these warnings.

I compile OpenMSX with:
```
CXX=clang++ make -j4 OPENMSX_FLAVOUR=debug
Build configuration:
Up to date: derived/x86_64-linux-debug/config/Version.ii
  Platform: x86_64-linux
  Flavour:  debug
  Compiler: clang++
  Subset:   full build
```

My clang version:
```
clang version 17.0.6 (Fedora 17.0.6-2.fc39)
Target: x86_64-redhat-linux-gnu
Thread model: posix

```

**Pull request merged**